app/controllers/adopters_controller.rb

```diff
@@ -176,6 +176,7 @@ def adopter_params
                                     ],
                                     references_attributes:
                                     [
+                                      :id,
```

Comment:
Since you're not modifying the id value of the reference, I don't think this line is required.

Reply:
I tried without id, but it inserted instead of updating.